File: MixAndMatch/MixAndMatch/Legs.cpp
#include "Legs.h"

Legs::Legs()
{
	addParts();
}


Legs::~Legs()
{
}

pair<string, gcroot<Bitmap^>> Legs::cratePair(string s, gcroot<Bitmap^> b)
{
	return make_pair(s, b);
}

void Legs::addParts()
{
	dictionary.insert(cratePair("fairy", createBitmap("Fairy_2.png")));
	dictionary.insert(cratePair("frankenstein", createBitmap("Frankenstein_2.png")));
	dictionary.insert(cratePair("skeleton", createBitmap("Skeleton_2.png")));
	dictionary.insert(cratePair("vampire", createBitmap("Vampire_2.png")));
	dictionary.insert(cratePair("werewolf", createBitmap("Werewolf_2.png")));
	dictionary.insert(cratePair("witch", createBitmap("Witch_2.png")));
}

gcroot<Bitmap^> Legs::createBitmap(string file)
{
	gcroot<Bitmap^> bitmap;
	bitmap->FromFile("C:/Users/´÷Àû·á/Documents/Visual Studio 2015/Projects/MixAndMatch/Pictures/" + gcnew String(file.c_str()));
	return bitmap;
}

File: MinAndMatchCSharp/MinAndMatchCSharp/Legs.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MinAndMatchCSharp
{
    class Legs : BodyParts
    {
        public Legs()
        {
            addParts();
        }

        protected override void addParts()
        {
            dictionary.Add("fairy", createBitmap("Fairy_2.png"));
            dictionary.Add("frankenstein", createBitmap("Frankenstein_2.png"));
            dictionary.Add("skeleton", createBitmap("Skeleton_2.png"));
            dictionary.Add("vampire", createBitmap("Vampire_2.png"));
            dictionary.Add("werewolf", createBitmap("Werewolf_2.png"));
            dictionary.Add("witch", createBitmap("Witch_2.png"));
        }
    }
}
